google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuerySQLException.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * BigQuery service exception.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/troubleshooting-errors">Google Cloud BigQuery
+ *     error codes</a>
+ */
+public final class BigQuerySQLException extends SQLException {
+
+  private static final long serialVersionUID = -5006625989225438209L;
+  private final List<BigQueryError> errors;
+
+  public BigQuerySQLException() {
+    this.errors = null;
+  }
+
+  public BigQuerySQLException(List<BigQueryError> errors) {
```

Comment:
If you end up using storage API under the hood, you may need to augment this to deal with grpc Status errors or having a conversion mechanism to make them BigQueryError.